src/node_contextify.cc

```diff
@@ -1398,6 +1401,22 @@ constexpr std::array<std::string_view, 3> esm_syntax_error_messages = {
     "Unexpected token 'export'",                     // `export` statements
     "Cannot use 'import.meta' outside a module"};    // `import.meta` references
 
+// TODO(joyeecheung): see if we can upstream an API to V8 for this.
```

Comment:
Ideally there would be a V8 API that tells us if there’s any ESM syntax anywhere in the file, rather than breaking on the first syntax error that it finds, to fix #50917.